src/parser/name_resolution.c

```diff
@@ -1201,74 +1216,82 @@ pt_bind_scope (PARSER_CONTEXT * parser, PT_BIND_NAMES_ARG * bind_arg)
 	    {
 	      assert (spec->info.spec.derived_table_type == PT_DERIVED_DBLINK_TABLE);
+
+	      int err;
+	      REMOTE_COLS *remote;
+	      PT_DBLINK_INFO *dblink_table = &table->info.dblink_table;
+
 	      if (table->info.dblink_table.is_name && table->info.dblink_table.url == NULL)
 		{
-		  int err;
-		  S_REMOTE_TBL_COLS *rmt_tbl_cols;
-		  REMOTE_COLS *remote;
-		  PT_DBLINK_INFO *dblink_table = &table->info.dblink_table;
-
 		  if (pt_resolve_dblink_server_name (parser, table, NULL) != NO_ERROR)
 		    {
 		      return;
 		    }
+		}
 
-		  /* remote table's column list */
-		  rmt_tbl_cols = new (std::nothrow) S_REMOTE_TBL_COLS;
+	      /* remote table's column list */
+	      rmt_tbl_cols = new (std::nothrow) S_REMOTE_TBL_COLS;
+	      if (rmt_tbl_cols == NULL)
+		{
+		  PT_ERRORf (parser, table, "Failed to get column information for memory allocation error", ER_DBLINK);
+		  return;
+		}
 
-		  err = pt_dblink_table_get_column_defs (parser, table, rmt_tbl_cols);
+	      err = pt_dblink_table_get_column_defs (parser, table, rmt_tbl_cols);
 
-		  if (table->info.dblink_table.remote_table_name && *table->info.dblink_table.remote_table_name)
+	      if (table->info.dblink_table.remote_table_name && *table->info.dblink_table.remote_table_name)
+		{
+		  /* error from table-name@server */
+		  if (err < 0)
 		    {
-		      /* error from table-name@server */
-		      if (err < 0)
+		      if (dblink_table->owner_name)
 			{
-			  if (dblink_table->owner_name)
-			    {
-			      PT_ERRORf4 (parser, table,
-					  "Failed to get column information for table [%s] on remote [%s].[%s]. err=%d",
-					  dblink_table->remote_table_name,
-					  dblink_table->owner_name->info.name.original,
-					  dblink_table->conn->info.name.original, err);
-			    }
-			  else
-			    {
-			      PT_ERRORf3 (parser, table,
-					  "Failed to get column information for table [%s] on remote [%s]. err=%d",
-					  dblink_table->remote_table_name, dblink_table->conn->info.name.original, err);
-			    }
-
-			  return;
+			  PT_ERRORf4 (parser, table,
+				      "Failed to get column information for table [%s] on remote [%s].[%s]. err=%d",
+				      dblink_table->remote_table_name,
+				      dblink_table->owner_name->info.name.original,
+				      dblink_table->conn->info.name.original, err);
 			}
-
-		      if (pt_remake_dblink_select_list (parser, &spec->info.spec, rmt_tbl_cols) != NO_ERROR)
+		      else
 			{
-			  return;
+			  PT_ERRORf3 (parser, table,
+				      "Failed to get column information for table [%s] on remote [%s]. err=%d",
+				      dblink_table->remote_table_name, dblink_table->conn->info.name.original, err);
 			}
+
+		      goto error_exit;
 		    }
 
-		  /* error from dblink(server, 'select ... from ...' */
-		  if (err < 0)
+		  if (table->info.dblink_table.cols == NULL)
 		    {
-		      return;
+		      if (pt_remake_dblink_select_list (parser, &spec->info.spec, rmt_tbl_cols) != NO_ERROR)
+			{
+			  return;
```

Comment:
This line must be 'goto error_exit;' , instead of 'return' to free rmt_tbl_cols.

Reply:
Ok, good point.